### ISSUE TITLE
Invalid report about "Possible unbalanced tuple unpacking"

### ISSUE THREAD

**Issue by toelke** [PyCQA/pylint#2621]
### Steps to reproduce

```
def _fn_a():
    return _fn_b() + (3, )

def _fn_c():
    vp1 = 1
    if vp1 < 0:
        vp1 = -vp1

    return vp1

def _fn_b():
    vp1 = _fn_c()
    return 1, vp1


A1, B1, C1 = _fn_a()
```

### Current behavior

> > pylint a.py
> ************* Module a
> a.py:1:0: C0111: Missing module docstring (missing-docstring)
> a.py:16:0: W0632: Possible unbalanced tuple unpacking with sequence: left side has 3 label(s), right side has 4 value(s) (unbalanced-tuple-unpacking)
> 
> ------------------------------------------------------------------
> Your code has been rated at 8.18/10 (previous run: 4.55/10, +3.64)

### Expected behavior

`_fn_a` only returns 3 items. If you remove the `if` from line 5f, pylint does not find any errors anymore.

### pylint --version output
> > pylint --version
> pylint 2.3.0
> astroid 2.0.4
> Python 3.6.6 |Anaconda, Inc.| (default, Jun 28 2018, 11:27:44) [MSC v.1900 64 bit (AMD64)]

**Comment by PCManticore** [comment]
Thanks for the report!

**Comment by awsumbill** [comment]
Hello, I had the same issue. 

Type `# pylint: disable=unbalanced-tuple-unpacking` at the end of the line with the error.

Hope this works :)

**Comment by Eljees** [comment]
Still reproduces on pylint 4.0.6 / astroid 4.0.4, and the cause is in the inference rather than in the checker.

`_fn_b()` on its own infers to a two-element tuple, as expected:

```
Tuple(elts=[Const(int), Name(vp1)])
```

But `_fn_b() + (3, )` infers to a tuple of **four**:

```
Tuple(elts=[Const(int) l.13, Const(int) l.6, Const(int) l.0, Const(int) l.3])
```

`Name(vp1)` has been replaced by *both* of its possible values. `vp1 = _fn_c()`, and `_fn_c()` infers to two constants because of the `if`, so the concatenation splices both into the result instead of contributing one element.

That is why removing the `if` makes the warning go away, as you noted: `_fn_c()` then has a single inferred value and the tuple comes out with three elements.

```python
# _fn_c() with the if
[Const(int) l.6, Const(int) l.0]     # two possible values
# resulting length of _fn_b() + (3,)
4        # 1 + 2 + 1  instead of  1 + 1 + 1
```

So `unbalanced-tuple-unpacking` is reporting faithfully what it was handed — the length it receives really is 4. The fix belongs in astroid's inference of `Tuple.__add__`, which should produce one element per element of the operands rather than one per inferred value; when an element is ambiguous the result is arguably `Uninferable` rather than a longer tuple.

Reproduced with the snippet from the issue body, unmodified.

AI-assisted (LLM used for drafting); the runs above are mine.

**Comment by Boulea7** [comment]
I can reproduce this on Astroid main: tuple concatenation expands one ambiguous element into multiple tuple slots. I’ll add a regression that preserves the tuple arity and update sequence concatenation so an ambiguous element is represented as unknown.
